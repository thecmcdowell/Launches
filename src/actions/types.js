export const DELETE_FAVORITE = "DELETE_FAVORITE"
export const ADD_FAVORITE = "ADD_FAVORITE"
export const FETCH_LAUNCHES = "FETCH_LAUNCHES"
export const FETCH_LAUNCHES_SUCCESS = "FETCH_LAUNCHES_SUCCESS"
export const FETCH_LAUNCHES_FAIL = "FETCH_LAUNCHES_FAIL"
export const GET_LAUNCH_INFO = 'GET_LAUNCH_INFO'
export const GET_LAUNCH_INFO_SUCCESS = 'GET_LAUNCH_INFO_SUCCESS'
export const GET_LAUNCH_INFO_FAIL = 'GET_LAUNCH_INFO_FAIL'