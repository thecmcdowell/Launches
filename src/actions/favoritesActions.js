import { ADD_FAVORITE, DELETE_FAVORITE } from './types'

export const addFavorite = launch => {
    return {
        type: ADD_FAVORITE,
        launch,
    }
}

export const deleteFavorite = id => {
    return {
        type: DELETE_FAVORITE,
        id
    }
}