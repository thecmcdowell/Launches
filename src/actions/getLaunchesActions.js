import * as actions from './types'

export const getLaunchInfo = id => {
    return {
        type: actions.GET_LAUNCH_INFO,
        payload: {
            request: {
                url: `launch/${id}`
            }
        }
    }
}

export const getLaunchList = () => {
    return {
        type: actions.FETCH_LAUNCHES,
        payload: {
            request: {
                url: 'launch/next/10'
            }
        }
    }
}