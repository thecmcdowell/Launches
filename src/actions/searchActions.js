import * as actions from './types'

export const searchLaunches = (date) => {
    return {
        type: actions.SEARCH,
        payload: {
            request: {
                url: `launch?startdate=${date}`
            }
        }
    }
}