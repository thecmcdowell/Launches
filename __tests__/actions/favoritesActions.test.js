import { ADD_FAVORITE, DELETE_FAVORITE } from '../../src/actions/types'
import * as actions from '../../src/actions/favoritesActions'
import data from '../dummyLaunch'

describe('favoritesActions tests', () => {
    it('adds a launch as favorite', () => {
            expect(actions.addFavorite(data)).toMatchSnapshot()
        }),
        it('deletes a favorited launch', () => {
            const outcome = {
                type: 'DELETE_FAVORITE',
                id: 1431
            }
            expect(actions.deleteFavorite(1431)).toEqual(outcome)
        })
})