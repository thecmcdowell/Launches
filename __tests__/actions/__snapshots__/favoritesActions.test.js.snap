// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`favoritesActions tests adds a launch as favorite 1`] = `
Object {
  "launch": Object {
    "count": 10,
    "launches": Array [
      Object {
        "changed": "2018-12-11 10:55:48",
        "failreason": null,
        "hashtag": null,
        "holdreason": null,
        "id": 1431,
        "infoURL": null,
        "infoURLs": Array [],
        "inhold": 0,
        "isoend": "20181227T020700Z",
        "isonet": "20181227T020700Z",
        "isostart": "20181227T020700Z",
        "location": Object {
          "countryCode": "RUS",
          "id": 34,
          "infoURL": "https://en.wikipedia.org/wiki/Vostochny_Cosmodrome",
          "name": "Vostochny Cosmodrome, Siberia, Russian Federation",
          "pads": Array [
            Object {
              "agencies": Array [
                Object {
                  "abbrev": "RFSA",
                  "changed": "2017-02-21 00:00:00",
                  "countryCode": "RUS",
                  "id": 63,
                  "infoURL": null,
                  "infoURLs": Array [
                    "http://en.roscosmos.ru/",
                    "https://www.youtube.com/channel/UCOcpUgXosMCIlOsreUfNFiA",
                    "https://twitter.com/Roscosmos",
                    "https://www.facebook.com/Roscosmos",
                  ],
                  "name": "Russian Federal Space Agency (ROSCOSMOS)",
                  "type": 1,
                  "wikiURL": "http://en.wikipedia.org/wiki/Russian_Federal_Space_Agency",
                },
              ],
              "id": 170,
              "infoURL": "",
              "latitude": 51.884395,
              "longitude": 128.333932,
              "mapURL": "https://www.google.ee/maps/place/51°53'03.8\\"N+128°20'02.2\\"E/",
              "name": "Cosmodrome Site 1S, Vostochny Cosmodrome, Siberia, Russian Federation",
              "wikiURL": "https://en.wikipedia.org/wiki/Vostochny_Cosmodrome",
            },
          ],
          "wikiURL": "",
        },
        "lsp": Object {
          "abbrev": "KhSC",
          "changed": "2017-02-21 00:00:00",
          "countryCode": "RUS",
          "id": 96,
          "infoURL": "http://www.khrunichev.ru/main.php?lang=en",
          "infoURLs": Array [
            "http://www.khrunichev.ru/main.php?lang=en",
          ],
          "name": "Khrunichev State Research and Production Space Center",
          "type": 1,
          "wikiURL": "http://en.wikipedia.org/wiki/Khrunichev_State_Research_and_Production_Space_Center",
        },
        "missions": Array [
          Object {
            "agencies": null,
            "description": "Kanopus-V is a series of Russian Earth observation satellites. They are used for real-time monitoring of natural and man-made emergencies, detecting forest fires, monitoring of land and water resources.",
            "id": 1159,
            "name": "Kanopus-V №5 & Kanopus-V №6",
            "payloads": Array [
              Object {
                "id": 369,
                "name": "Kanopus-V №5, 6",
              },
              Object {
                "id": 370,
                "name": "Flock-w (12 x Dove)",
              },
              Object {
                "id": 372,
                "name": "GRUS",
              },
              Object {
                "id": 376,
                "name": "ZACube-2",
              },
              Object {
                "id": 377,
                "name": "UWE-4",
              },
              Object {
                "id": 380,
                "name": "Lume 1",
              },
              Object {
                "id": 381,
                "name": "Lemur-2",
              },
              Object {
                "id": 382,
                "name": "D-Star One (Sparrow)",
              },
              Object {
                "id": 383,
                "name": "D-Star One (iSat)",
              },
            ],
            "type": 1,
            "typeName": "Earth Science",
            "wikiURL": "",
          },
        ],
        "name": "Soyuz 2.1a/Fregat | Kanopus-V №5 & Kanopus-V №6",
        "net": "December 27, 2018 02:07:00 UTC",
        "netstamp": 1545876420,
        "probability": -1,
        "rocket": Object {
          "agencies": Array [],
          "configuration": "2.1a/Fregat",
          "familyname": "Soyuz",
          "id": 83,
          "imageSizes": Array [
            320,
            480,
            640,
            720,
            768,
            800,
            960,
            1024,
            1080,
            1280,
            1440,
            1920,
          ],
          "imageURL": "https://s3.amazonaws.com/launchlibrary/RocketImages/placeholder_1920.png",
          "infoURLs": Array [],
          "name": "Soyuz 2.1a/Fregat",
          "wikiURL": "https://en.wikipedia.org/wiki/Soyuz-2",
        },
        "status": 1,
        "tbddate": 0,
        "tbdtime": 0,
        "vidURL": null,
        "vidURLs": Array [],
        "westamp": 1545876420,
        "windowend": "December 27, 2018 02:07:00 UTC",
        "windowstart": "December 27, 2018 02:07:00 UTC",
        "wsstamp": 1545876420,
      },
      Object {
        "changed": "2018-12-07 17:45:49",
        "failreason": null,
        "hashtag": "#Iridium8",
        "holdreason": null,
        "id": 1276,
        "infoURL": null,
        "infoURLs": Array [],
        "inhold": 0,
        "isoend": "20190107T155300Z",
        "isonet": "20190107T155300Z",
        "isostart": "20190107T155300Z",
        "location": Object {
          "countryCode": "USA",
          "id": 18,
          "infoURL": "",
          "name": "Vandenberg AFB, CA, USA",
          "pads": Array [
            Object {
              "agencies": Array [],
              "id": 100,
              "infoURL": "",
              "latitude": 34.632,
              "longitude": -120.611,
              "mapURL": "http://maps.google.com/maps?q=34.632+N,+120.611+W",
              "name": "Space Launch Complex 4E, Vandenberg AFB, CA",
              "wikiURL": "",
            },
          ],
          "wikiURL": "",
        },
        "lsp": Object {
          "abbrev": "SpX",
          "changed": "2017-02-21 00:00:00",
          "countryCode": "USA",
          "id": 121,
          "infoURL": null,
          "infoURLs": Array [
            "http://www.spacex.com/",
            "https://twitter.com/SpaceX",
            "https://www.facebook.com/SpaceX",
            "https://www.youtube.com/channel/UCtI0Hodo5o5dUb67FeUjDeA",
          ],
          "name": "SpaceX",
          "type": 3,
          "wikiURL": "http://en.wikipedia.org/wiki/SpaceX",
        },
        "missions": Array [
          Object {
            "agencies": Array [
              Object {
                "abbrev": "IRDM",
                "changed": "2018-03-17 23:49:56",
                "countryCode": "USA",
                "id": 202,
                "infoURL": "",
                "infoURLs": Array [],
                "name": "Iridium Communications",
                "type": 3,
                "wikiURL": "https://en.wikipedia.org/wiki/Iridium_Communications",
              },
            ],
            "description": "SpaceX’s Falcon 9 rocket will deliver 10 satellites to low Earth orbit for Iridium, a global leader in mobile voice and data satellite communications. This is the eighth set of 10 satellites in a series of 75 total satellites that SpaceX will launch for Iridium’s next generation global satellite constellation, Iridium® NEXT.",
            "id": 602,
            "name": "Iridium-8",
            "payloads": Array [],
            "type": 10,
            "typeName": "Communications",
            "wikiURL": "https://en.wikipedia.org/wiki/Iridium_satellite_constellation#Next-generation_constellation",
          },
        ],
        "name": "Falcon 9 Block 5 | Iridium-8",
        "net": "January 7, 2019 15:53:00 UTC",
        "netstamp": 1546876380,
        "probability": -1,
        "rocket": Object {
          "agencies": Array [
            Object {
              "abbrev": "SpX",
              "changed": "2017-02-21 00:00:00",
              "countryCode": "USA",
              "id": 121,
              "infoURL": null,
              "infoURLs": Array [
                "http://www.spacex.com/",
                "https://twitter.com/SpaceX",
                "https://www.facebook.com/SpaceX",
                "https://www.youtube.com/channel/UCtI0Hodo5o5dUb67FeUjDeA",
              ],
              "name": "SpaceX",
              "type": 3,
              "wikiURL": "http://en.wikipedia.org/wiki/SpaceX",
            },
          ],
          "configuration": "9 Block 5",
          "familyname": "Falcon",
          "id": 188,
          "imageSizes": Array [
            320,
            480,
            640,
            720,
            768,
            800,
            960,
            1024,
            1080,
            1280,
            1440,
            1920,
          ],
          "imageURL": "https://s3.amazonaws.com/launchlibrary/RocketImages/Falcon9Block5.jpg_1920.jpg",
          "infoURL": "http://www.spacex.com/falcon9",
          "infoURLs": Array [
            "http://www.spacex.com/falcon9",
          ],
          "name": "Falcon 9 Block 5",
          "wikiURL": "https://en.wikipedia.org/wiki/Falcon_9_Full_Thrust#Block_5",
        },
        "status": 1,
        "tbddate": 0,
        "tbdtime": 0,
        "vidURL": null,
        "vidURLs": Array [
          "http://www.spacex.com/webcast",
        ],
        "westamp": 1546876380,
        "windowend": "January 7, 2019 15:53:00 UTC",
        "windowstart": "January 7, 2019 15:53:00 UTC",
        "wsstamp": 1546876380,
      },
    ],
    "offset": 0,
    "total": 183,
  },
  "type": "ADD_FAVORITE",
}
`;
